### DISCUSSION TITLE
How to modify source code on libxc package and recompile?

### DISCUSSION THREAD

**Discussion by rsielguea** [spack/spack#31017] [Q&A]
I need to modify "util.c" file located in the src/ directory on libxc. However, in the spack package there is no "util.c" file at all (not sure why since it is basic for the library to function). I tried to create my own libxc package but I get this error:

```
$ spack create http://www.tddft.org/programs/libxc/down.php?file=5.2.3/libxc-5.2.3.tar.gz
==> This looks like a URL for libxc
==> Warning: Spack was unable to fetch url list due to a certificate verification problem. You can try running spack -k, which will not check SSL certificates. Use this at your own risk.
==> Warning: Spack was unable to fetch url list due to a certificate verification problem. You can try running spack -k, which will not check SSL certificates. Use this at your own risk.
==> Found 1 version of libxc:
  
  5.2.3  http://www.tddft.org/programs/libxc/down.php?file=5.2.3/libxc-5.2.3.tar.gz

==> Something failed on http://www.tddft.org/programs/libxc/down.php?file=5.2.3/libxc-5.2.3.tar.gz, skipping.  (All fetchers failed for spack-stage-8yUU7L)
==> Error: Could not fetch any versions for libxc
```

So, in the existing package I can't modify the file I need and recompile, and if I try to make my own package spack does not fetch. I have tried to download the .tar.gz file, untar, and try to work around this but I get other types of errors and I can't move forward. 
Help please!

**Comment by haampie** [comment]
You have two options:
- Add a patch to the existing package
- Build from local sources using `spack develop`: https://spack-tutorial.readthedocs.io/en/latest/tutorial_developer_workflows.html#developer-workflows-tutorial
